src/services/jobs.js

```diff
@@ -3,185 +3,186 @@
+				const instanceType = (jobData.schedule.scheduleType || '').toLowerCase()
+
+				if (instanceType == common.ONCE && jobData.schedule.interval && jobData.schedule.interval != '') {
+					const newInstance = await createJobInstanceForOnce(jobData.name, jobData.schedule.interval)
+					return newInstance
+				} else if (
+					instanceType == common.EVERY &&
+					jobData.schedule.interval &&
+					jobData.schedule.interval != ''
+				) {
+					const newInstance = await createJobInstanceForEvery(jobData.name, jobData.schedule.interval)
+					return newInstance
+				} else {
+					const newInstance = await createJobInstanceForNow(jobData.name)
+					return newInstance
+				}
+			}
+		} else {
+			return {
+				message: responseMessage.JOB_DEFINITION_ALREADY_EXISTS,
+				success: false,
+				status: httpResponse.BAD_REQUEST,
+			}
+		}
+	} catch (err) {
+		return {
+			message: err,
+			success: false,
+			status: httpResponse.BAD_REQUEST,
+		}
+	}
+}
 
 /**
-    * create new job instance that exicute once
-    * @method
-    * @name jobInstanceForOnce
-    * @param {Object} jobName -name of job definition.
-    * @param {Object} scheduleTime -job exicution time.
-    * @returns {JSON} - returns job instance creation details.
-*/
-
-const createJobInstanceForOnce = async ( jobName, scheduleTime ) => {
-    try {
-        const job = {
-            name : jobName,
-            interval : scheduleTime
-        };
-        //get all jobs definition from DB
-        const jobs = await jobsReady;
-        //checking job Definition is present
-        const jobExists = await utils.checkForDuplicateJobDefinition( job, jobs );
-
-        if ( jobExists > 0 ) {
-            const scheduleJob = await scheduleOnce( job, agenda );
-            return scheduleJob;
-        } else {
-            return ({
-                message : responseMessage.JOB_DEFINITION_NOT_FOUND,
-                success : false,
-                status : httpResponse.BAD_REQUEST
-            });
-        }
-    }catch(err){
-        return ({
-                message : err,
-                success : false,
-                status : httpResponse.BAD_REQUEST
-            });
-    }
-};
+ * create new job instance that exicute once
+ * @method
+ * @name jobInstanceForOnce
+ * @param {Object} jobName -name of job definition.
+ * @param {Object} scheduleTime -job exicution time.
+ * @returns {JSON} - returns job instance creation details.
+ */
+
+const createJobInstanceForOnce = async (jobName, scheduleTime) => {
+	try {
+		const job = {
+			name: jobName,
+			interval: scheduleTime,
+		}
+		//get all jobs definition from DB
+		const jobs = await jobsReady
+		//checking job Definition is present
+		const jobExists = await utils.checkForDuplicateJobDefinition(job, jobs)
+
+		if (jobExists > 0) {
+			const scheduleJob = await scheduleOnce(job, agenda)
+			return scheduleJob
+		} else {
+			return {
+				message: responseMessage.JOB_DEFINITION_NOT_FOUND,
```

Comment:
@VISHNUDAS-tunerlabs response should come from common function, check mentoring service